File: solutions/csharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.cs
class Lasagna
{
    private const int ExpectedMinutes = 40;
    private const int LayerPrepTime = 2;
    
    public int ExpectedMinutesInOven()
    {
        return ExpectedMinutes;
    }
    
    public int RemainingMinutesInOven(int minutesInOven)
    {
        return ExpectedMinutes - minutesInOven;
    }
    
    public int PreparationTimeInMinutes(int layers)
    {
        return layers * LayerPrepTime;
    }
    
    public int ElapsedTimeInMinutes(int layers, int minutesInOven)
    {
        return minutesInOven + PreparationTimeInMinutes(layers);
    }
}

File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna
let expectedMinutesInOven = 40
let remainingMinutesInOven x =
    expectedMinutesInOven - x
let preparationTimeInMinutes layers =
    layers * 2
let elapsedTimeInMinutes layers timeInOven =
    (preparationTimeInMinutes layers) + timeInOven